core/trietxn.go

```diff
@@ -0,0 +1,68 @@
+package core
+
+import (
+	"github.com/bits-and-blooms/bitset"
+	"github.com/dgraph-io/badger/v3"
+)
+
+// TrieTxn is a database transaction on a trie.
+type TrieTxn struct {
+	badgerTxn *badger.Txn
+	prefix    []byte
+}
+
+func NewTrieTxn(badgerTxn *badger.Txn, prefix []byte) *TrieTxn {
+	return &TrieTxn{
+		badgerTxn: badgerTxn,
+		prefix:    prefix,
+	}
+}
+
+// dbKey creates a byte array to be used as a key to our KV store
+// it simply appends the given key to the configured prefix
+func (txn *TrieTxn) dbKey(key *bitset.BitSet) ([]byte, error) {
```

Comment:
Please change the `func (txn *TrieTxn)` -> `func (t *TrieTxn)` to follow go conventions for all the functions.